src/parsy/__init__.py

```diff
@@ -28,26 +28,37 @@ def line_info(self):
             return '<out of bounds index {!r}>'.format(self.index)
 
     def __str__(self):
-        return 'expected {} at {}'.format(self.expected, self.line_info())
+        expected_list = [repr(e) for e in self.expected]
+        expected_list.sort()
```

Comment:
`expected_list = sorted(repr(e) for e in self.expected)` or even `sorted(map(repr, self.expected))`